### ISSUE TITLE
Conformal nonconformity score is nearly constant (0.72–0.80 for everything), so sets cannot widen under shift

### ISSUE THREAD

**Issue by johnson2006christopher** [johnson2006christopher/adaptshot#86]
## What

`ConformalEngine.softmax_nonconformity` has almost no dynamic range. Measured on a 5-way PlantVillage episode, the minimum-over-classes nonconformity of a query is:

| query | score |
|---|---|
| clean, in-distribution | 0.772 |
| the same photographs, Gaussian blur σ=4 | 0.785 |
| a healthy **tomato** leaf — a crop the model was never shown | 0.757 |

Every calibration score in that episode fell between 0.736 and 0.794. A different crop scores *lower* than a clean in-distribution leaf.

## Why

```python
max_dist = np.max(distances) + 1e-8
logits = -distances / max_dist   # every logit is in [-1, 0]
```

Five logits confined to [−1, 0] give a softmax that is close to uniform (each ≈ 0.2), so `1 − p(true)` is ≈ 0.76–0.80 whatever the distances are. The score records *which* class is nearest by a hair; it cannot record that the query is far from **all** of them.

## Why it matters — this is the mechanism of #29's silent failure

The shift pilot shows coverage falling from 96.7% to ~78% under strong blur, JPEG or downscale **while the mean set size stays flat at ~1.5**. A conformal set can only widen if the score for a shifted query exceeds the calibration quantile; with a score that doesn't move, it never does. The guarantee fails and nothing in the set says so. The OOD flag is the only thing that rises, and it is a separate mechanism.

It also explains why conformal over-covers on clean data (97.5% at a 90% target, #18): the quantile is a razor-thin slice of a near-constant distribution, and "always include the top-1" is doing most of the covering.

## What to try

- A temperature: `logits = -distances / (T · scale)` with T fitted on the calibration split, or the distances *standardised* (subtract the mean, divide by the std across classes) rather than divided by the max.
- Or a distance-based score, which the engine already supports (`score_method="distance"`) but the learner does not use by default.
- Measure both with `tests/test_conformal_coverage.py` (synthetic) **and** `benchmarks/run_shift.py` (real photographs under shift): the target is sets that widen under shift while coverage and size on clean data stay comparable.

## Not fixed in #29's PR

Changing the score moves every published conformal figure. It needs its own PR with the before/after on both harnesses.

**Comment by johnson2006christopher** [comment]
Done in #90, merged into `v0.3.0`.

**The finding held up.** The max-scaled softmax scored a clean photograph 0.772, the same photograph blurred 0.785, and a leaf of a crop the model had never seen 0.757 — no dynamic range at all, because five logits confined to [−1, 0] give a near-uniform softmax.

**The fix.** The default score is now the distance ratio `d_true / min(d)`, through one dispatcher, `ConformalEngine.nonconformity`, that every caller uses so calibration and prediction cannot score differently — for one commit they did, and the synthetic harness failed thirteen cells, which is what it is for. `softmax` and `distance` remain selectable.

**Measured.** PlantVillage 5-way 5-shot, 100 episodes, α = 0.1: coverage **98.1% ± 0.6** at mean set size **1.66** (was 97.5% at 2.05) — tighter sets, higher coverage, accuracy unchanged; every cell of the synthetic guarantee harness passes. Under shift (40 episodes) the sets now widen where before they could not — mean size 1.34 → 2.13 at blur σ=4 — and coverage at the worst levels recovers from ~78% to ~86% before corrections and ~90% after. It still falls short of the 90% target without corrections: the quantile was set on clean photographs and cannot know the queries moved. That is exchangeability breaking, not the score, and the README's shift section now says so with the new numbers rather than the old "silent failure" framing, which the new numbers no longer support.

Stated for the record: `ConformalEngine` is stable and this changes its default score — set contents move — without removing or renaming anything. I called it a bug fix because the old default was defective by measurement; it is a decision you can reverse by name.
